app/code/Kogut/ProductsGeneration/Model/Service/CreateProduct.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kogut\ProductsGeneration\Model\Service;
+
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Api\CategoryLinkManagementInterface;
+
+class CreateProduct
+{
+    /**
+     * @var ProductFactory
+     */
+    private $productFactory;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var CategoryLinkManagementInterface
+     */
+    private $linkManagement;
+
+    /**
+     * CreateProduct constructor.
+     * @param ProductFactory $productFactory
+     * @param ProductRepositoryInterface $productRepository
+     * @param CategoryLinkManagementInterface $linkManagement
+     */
+    public function __construct(
+        ProductFactory $productFactory,
+        ProductRepositoryInterface $productRepository,
+        CategoryLinkManagementInterface $linkManagement
+    ) {
+        $this->productFactory = $productFactory;
+        $this->productRepository = $productRepository;
+        $this->linkManagement = $linkManagement;
+    }
+
+    /**
+     * Create test simple product service
+     * @param $categoryId
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Exception\StateException
+     */
+    public function createSimpleProduct($categoryId)
+    {
+        $product = $this->productFactory->create();
+        $productSku = 'testSku' . microtime();
+        $product->setSku($productSku);
+        $product->setName('testName' . microtime());
+        $product->setAttributeSetId(4);
```

Comment:
Please use $product->getDefaultAttributeSetId() instead of numeric constants
Try to use named constants everywhere you can

Reply:
Fixed